src/game/Spells/Spell.cpp

```diff
@@ -5082,7 +5082,9 @@ void Spell::TakeAmmo() const
 {
     if (m_attackType == RANGED_ATTACK && m_caster->GetTypeId() == TYPEID_PLAYER)
     {
-        Item* pItem = ((Player*)m_caster)->GetWeaponForAttack(RANGED_ATTACK, true, false);
+        auto player = (Player*)m_caster;
```

Comment:
Why would you use `auto` for a trivial datatype? 🤐

Reply:
Why not? Using it on non trivial type is better? Reading that line you know directly what type will be assigned. Devs are lazy and there is 3 keypress less in "auto" than "Player*". Almost 50% less typing... Iam not so against :p

Reply:
I guess it just comes down to me and how I work with pointers, references and how I keep them in my head. I know that many people have issues with keeping track of it(C# does not have this).  If its 1.5-2.x bigger its a non issue, when its something like std::vector<std::map<>>.... the tool to use is typedef instead, again, at least for me. The reality is that for a dev which is 100% dev in his time, this is non-issue, but most people contributing are not, hence its always great when they have to write that asterisk.

Also now that I think about it, the bigger thing here is that C-Style cast, should be static_cast instead.

Reply:
I used auto because I am used to it. At work we have modern code bases (C++17) and there we mostly stick to the IsoCpp core guidelines which encourage auto in most cases - there are exceptions but usually you can trust the compiler's ability to infer types. I know this code base isn't 'modern', and the issue was not about refactoring a function, but adding functionality instead, which is why I left it that way. To me it is apparent that a variable called `player` is an instance of `Player`. I definitely did not want to add yet another line of `(Player*)m_caster)->` to that function. 😅

Reply:
No. We decided against adding it, because it uses reinterpret_cast and we deemed it unnecessary.